includes/class-twitter-accounts.php

```diff
@@ -0,0 +1,331 @@
+			$url = $this->twitter_api->get_authorize_url( $request_token['oauth_token'] );
+			if ( ! empty( $url ) ) {
+				wp_redirect( $url ); // phpcs:ignore WordPress.Security.SafeRedirect.wp_redirect_wp_redirect
+				exit();
+			}
+		} catch ( \Exception $e ) {
+			$error   = $e->getMessage();
+			$decoded = json_decode( $e->getMessage() );
+			if ( json_last_error() === JSON_ERROR_NONE ) {
+				$error = $decoded;
+			}
+
+			if ( ! empty( $error->errors ) ) {
+				$error         = current( $error->errors );
+				$error_message = $error->message;
+			} elseif ( ! empty( $error ) ) {
+				$error_message = $error;
+			} else {
+				$error_message = __( 'Something went wrong. Please try again.', 'autoshare-for-twitter' );
+			}
+
+			$this->set_connection_notice( 'error', $error_message );
+		}
+
+		// Redirect back to AutoShare settings page.
+		wp_safe_redirect( admin_url( 'options-general.php?page=autoshare-for-twitter' ) );
+		exit();
+	}
+
+	/**
+	 * Callback for Twitter authorization process.
+	 *
+	 * @throws \Exception If the request token is not valid.
+	 */
+	public function authoshare_authorize_user_callback() {
+		// Check if the user has the correct permissions.
+		if ( ! current_user_can( 'manage_options' ) ) {
+			wp_die( esc_html__( 'You do not have sufficient permissions to access this page.', 'autoshare-for-twitter' ) );
+		}
+
+		try {
+			// Check if users has denied the authorization.
+			// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$is_deniend = isset( $_GET['denied'] ) ? sanitize_text_field( wp_unslash( $_GET['denied'] ) ) : false;
+			if ( $is_deniend ) {
+				throw new \Exception( __( 'You have denied the authorization.', 'autoshare-for-twitter' ) );
+			}
+
+			$oauth_token        = isset( $_COOKIE['autoshare_oauth_token'] ) ? sanitize_text_field( wp_unslash( $_COOKIE['autoshare_oauth_token'] ) ) : '';
+			$oauth_token_secret = isset( $_COOKIE['autoshare_oauth_token_secret'] ) ? sanitize_text_field( wp_unslash( $_COOKIE['autoshare_oauth_token_secret'] ) ) : '';
+
+			// Check if the request token is valid.
+			if ( ! isset( $_REQUEST['oauth_token'] ) || sanitize_text_field( wp_unslash( $_REQUEST['oauth_token'] ) !== $oauth_token ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+				wp_die( 'Something went wrong. Please try again.' );
+			}
+
+			// Get the access token.
+			$oauth_verifier = isset( $_REQUEST['oauth_verifier'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['oauth_verifier'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$access_token   = $this->twitter_api->get_access_token( $oauth_token, $oauth_token_secret, $oauth_verifier );
+
+			if ( ! $access_token || ! isset( $access_token['oauth_token'] ) || ! isset( $access_token['oauth_token_secret'] ) ) {
+				throw new \Exception( 'autoshare_twitter_authorize_error', __( 'Something went wrong during getting access token. Please try again', 'autoshare-for-twitter' ) );
```

Comment:
This can be "Something went wrong while generating the access token" or "Something went wrong while fetching the access token"